slashing-protection/src/main/resources/migrations/postgresql/V00012__add_highwatermark_metadata.sql

```diff
@@ -0,0 +1,63 @@
+ALTER TABLE metadata
+    ADD COLUMN high_watermark_epoch NUMERIC(20),
+    ADD COLUMN high_watermark_slot NUMERIC(20);
+
+-- inserted high watermark should be above low watermark
+
+CREATE OR REPLACE FUNCTION check_high_watermarks() RETURNS TRIGGER AS $$
+DECLARE
+    max_slot NUMERIC(20);
+    max_epoch NUMERIC(20);
+BEGIN
+SELECT MAX(slot) INTO max_slot FROM low_watermarks;
```

Comment:
Will doing this have an impact on performance when there are a large number of validators? There is no index on the slot or epochs so postgresql will probably have to do a table scan I think.

Reply:
Good catch, I haven't measured the performance of this yet.

Reply:
Tested this up to 100,000 validators. 

Seq Scan does occur, although only 780 rows interestingly (and this row number doesn't change with the number of validators).

Max time was 49ms so don't think there's a performance issue.

https://github.com/Consensys/web3signer/pull/896#issuecomment-1710829153